src/views/charts/TradingRewardsChart.tsx

```diff
@@ -91,18 +91,19 @@ export const TradingRewardsChart = ({
     SELECTED_PERIOD.name
   );
 
-  const rewardsData = useMemo(
-    () =>
-      periodTradingRewards && canViewAccount
-        ? periodTradingRewards.reverse().map(
-            (datum): TradingRewardsDatum => ({
-              date: new Date(datum.endedAtInMilliseconds).valueOf(),
-              cumulativeAmount: datum.cumulativeAmount,
-            })
-          )
-        : [],
-    [periodTradingRewards, canViewAccount]
-  );
+  const rewardsData = useMemo(() => {
+    if (periodTradingRewards && canViewAccount) {
+      const res = periodTradingRewards.map(
+        (datum): TradingRewardsDatum => ({
+          date: new Date(datum.endedAtInMilliseconds).valueOf(),
+          cumulativeAmount: datum.cumulativeAmount,
+        })
+      );
+      res.sort((datumA, datumB) => datumA.date - datumB.date);
```

Comment:
are we worried about the order given to us being wrong? I think doing map then reverse is fine because map will create a new array every time.

Reply:
oh it was my fault, before we were calling toArray first, which was creating new arrays

https://github.com/dydxprotocol/v4-web/pull/653/files#diff-aee1b9500dfa4a2505c736c038fb862bff2346ab38e4f50fce46877b0c4e7b28L97

Reply:
Honestly, why don't we switch to toReversed everywhere and add a lint rule to ban .reverse

Reply:
https://linear.app/dydx/issue/OTE-482/[web-v4]-ban-reverse

sounds good, made a ticket for it

Reply:
wait... `toArray` is something we use on kotlin objects. does the `.map` function play well with kotlin lists? 👀 
cc: @moo-onthelawn

Reply:
hmm it works for me both in empty and non empty data cases!

Reply:
synced offline: 
- [tyler's PR calls `.toArray` in the selector](https://github.com/dydxprotocol/v4-web/pull/653/files#diff-6fab05d9743dd1efcda3ef53b0313d3f6a4f5a7a30760cf854e5e7b1b1f00c18R662) so it's g2g

Reply:
ok attempted a minor patch of using .toReversed() but I think i need to double check our compiler options in vercel because it fails with `src/views/charts/TradingRewardsChart.tsx(102,18): error TS2551: Property 'toReversed' does not exist on type 'TradingRewardsDatum[]'. Did you mean 'reverse'?` (while it works locally) // gonna punt to the linear task Jeremy made as follow up for the time being!